core/src/main/java/jenkins/model/GlobalConfiguration.java

```diff
@@ -75,4 +76,13 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
         return Jenkins.get().getDescriptorList(GlobalConfiguration.class);
         // pointless type parameters help work around bugs in javac in earlier versions http://codepad.org/m1bbFRrH
     }
+
+    /**
+     * Assume by default that GlobalConfiguration items require Jenkins.ADMINISTER permission
+     * Ovveride to return something different if appropriate
```

Comment:
what did the code used to be?

Reply:
it didn't exist prior to this, so there was no way to query a GlobalConfiguration item to see what permission it requires or checks for.  This meant you had to try and load it and find out if it threw an exception or not (which gives you the error shown in the ux screenshot).